### DISCUSSION TITLE
Is there a way to call/repeat the same endpoint multiple times per "Tag" or when tags are not present?

### DISCUSSION THREAD

**Discussion by proper-function** [apideck-libraries/portman#456] [Q&A] [answered]
First off, this library is amazing! I found it a few days ago and have been using it to automate our postman tests. 

I'm wondering if it's possible to run the same request multiple times per Tag. For example, I am testing CRUD and would like to run getById multiple times to check things like if the update actually changed the value, or if the delete was successful.

Eg, instead of;
POST
UPDATE
GET
DELETE
GET ALL

I'm wondering if this is possible
POST
GET
UPDATE
GET
DELETE
GET
GET ALL

In my case, many tags are dependent on one endpoint, so I used openapi-filter to strip the tags so everything is listed without any folders. Then I used the orderOfOperations to specify the exact order of every endpoint. So it is working well enough to use but I'm still curious.

**Comment by thim81** [comment, marked as answer]
Hi @proper-function 

I don't think we support the case to "copy" operations, there is a workaround that you might experiment with although I'm not a big fan of the method: `postman.setNextRequest()`. You could leverage `extendTests` to inject `postman.setNextRequest()` in the PUT/POST operations, but again it is complicated.

An alternative could be to create multiple test suites in different Postman collections using `orderOfOperations` in combination with the build-in filtering options: https://github.com/apideck-libraries/portman/blob/main/examples/cli-filtering/readme.md#portman-cli-options, which would result in multiple collections like:

Postman A collection:
POST
GET

Postman B collection:
PUT
GET

Postman C collection:
DELETE
GET
GET ALL

The "copy" request sounds like a nice new feature, but it would require some more thought.
Portman allows "overwrite" and "pre-request" injection, if you would copy a "request", what would you expect? Is it a copy of the original request, the modified request, or something else? And how to organize the requests? We have "orderOfOperations" for ordering requests which are already quite complex.

@nicklloyd What do you think about the "copy request" option?

**Reply by proper-function** [reply]
Hi @thim81, 

I am thinking the multiple collections is what I'll end up doing if my current tests get more complicated. 

I think my ideal is way outside of the scope of portman where it wouldn't really be a "copy" request. Rather a more advanced "orderOfOperations" where the folder structure and endpoint calls could be specified in advance and an ID given to them (apart from operation id). This would allow duplicate requests to the openapi spec endpoints or any other api. Then there could be a definition section where each ID could specify any attachVariables, overrides, tests, etc that weren't globally specified. 

I'm a developer though, so I know what it's like to roll your eyes at someone's impossible standards xD. Realistically, I could probably do something with --localPostman to get the custom folder and endpoint structure. Or just do some rearranging post portman generation. It does its job very well and I commend you guys for that!

**Comment by nicklloyd** [comment]
Hi sorry guys - am just seeing this thread now. Is this not something you could achieve via the [Variation Testing in Portman](https://github.com/apideck-libraries/portman/tree/main/examples/testsuite-variation-tests) or (am also just seeing we still don't have docs for) integrationTests?

**Reply by nicklloyd** [reply]
Have a look at the Kitchen Sink though it also needs some updating, there is an example of implementing integration tests into grouped 'runs' https://github.com/apideck-libraries/portman/blob/04db80c6455715ebe159edf465b3c49405ff351e/__tests__/fixtures/portman.kitchensink.json

**Reply by proper-function** [reply]
Hi @nicklloyd

The integrations tests look like they do exactly what I need! I'll have to try them out. I think the only thing missing would be custom endpoint calls...authentication in my case. I have a pre-test script that should work for that though.
